src/cc/libbpf.c

```diff
@@ -177,9 +187,14 @@ int bpf_prog_load(enum bpf_prog_type prog_type,
   if (ret < 0 && !log_buf) {
     // caller did not specify log_buf but failure should be printed,
     // so call recursively and print the result to stderr
+    buffer_size = attr.insn_cnt * 128;
```

Comment:
I think duplicating exponential increase logic is a lesser evil than passing a pointer to allocated memory from c side into python.